tests/ctl/core/test_system.py

```diff
@@ -7,44 +7,47 @@
 from py._path.local import LocalPath
 
 from fides.ctl.connectors.models import OktaConfig
-from fides.ctl.core import api
 from fides.ctl.core import system as _system
+from fides.ctl.core.api import generate_resource_url
 from fides.ctl.core.config import FidesConfig
 
 
-def create_server_systems(test_config: FidesConfig, systems: List[System]) -> None:
+def create_server_systems(
+    test_config: FidesConfig, systems: List[System], test_client
+) -> None:
     for system in systems:
-        api.create(
-            url=test_config.cli.server_url,
-            resource_type="system",
-            json_resource=system.json(exclude_none=True),
+        test_client.post(
+            generate_resource_url(test_config.cli.server_url, "system"),
             headers=test_config.user.request_headers,
+            data=system.json(exclude_none=True),
         )
 
 
-def delete_server_systems(test_config: FidesConfig, systems: List[System]) -> None:
+def delete_server_systems(
+    test_config: FidesConfig, systems: List[System], test_client
+) -> None:
     for system in systems:
-        api.delete(
-            url=test_config.cli.server_url,
-            resource_type="system",
-            resource_id=system.fides_key,
+        test_client.delete(
+            generate_resource_url(
+                test_config.cli.server_url, "system", resource_id=system.fides_key
+            ),
             headers=test_config.user.request_headers,
         )
 
 
 @pytest.fixture(scope="function")
 def create_test_server_systems(
-    test_config: FidesConfig, redshift_systems: List[System]
+    test_config: FidesConfig, redshift_systems: List[System], test_client
 ) -> Generator:
     systems = redshift_systems
-    delete_server_systems(test_config, systems)
-    create_server_systems(test_config, systems)
+    delete_server_systems(test_config, systems, test_client)
+    create_server_systems(test_config, systems, test_client)
     yield
-    delete_server_systems(test_config, systems)
+    delete_server_systems(test_config, systems, test_client)
```

Comment:
This fixture is an example using the`test_client` directly which is aware of test_mode, instead of using the `api.create`/`api.delete` methods so I believe this fixture is creating the resources on the test db.  I made similar changes across several tests.

I've updated the CTL API to inject sessions into the routes, which seems to be working well.  But I think separate methods in src/fides/ctl/core/api.py are still talking to a webserver that's using the application db.   I'm not sure how to resolve yet -  will look more tomorrow - The test `test_get_all_server_systems` is failing because it's probably looking for the resources in the application db because it's calling methods in core/api.  There are a couple of other files that aren't failing per se, but are also writing to the application db.